### PR TITLE
Add access policy enum

### PR DESCRIPTION
**Pull request by fiedosiukr**
This PR moves hard-coded values of access policy to an enum.

### CODE REVIEW

**Comment by alistairewj** [comment]
@tompollard this is a pretty simple / nice QOL improvement that might be nice to merge before we rebase the other PRs (particularly before the one where we add a new access policy). it's more line changes than it needs to be because of the flake8 formatting, but mostly it's fairly straightforward

**Comment by tompollard** [comment]
@alistairewj yeah, i agree this is a nice improvement. i briefly reviewed it already but just wanted to do some testing before merging. will take a look this morning.

**Comment by tompollard** [comment]
@fiedosiukr please could you rebase?

**Comment by tompollard** [comment]
@fiedosiukr thanks for looking at this. In case you don't get the pings from our testing server, it looks like some of the tests are failing. Let me know if you can't see the details and I'll try to set the permissions so that you can. 

[Update: posted test output below]

```
0s
Run cd physionet-django
  cd physionet-django
  python manage.py makemigrations --dry-run --no-input --check
  python manage.py resetdb
  python manage.py loaddemo
  coverage run --source='.' manage.py test --verbosity=3 --keepdb
  coverage report -m
  shell: bash --noprofile --norc -e -o pipefail {0}
  env:
    DJANGO_SETTINGS_MODULE: physionet.settings.settings
    TEST_GCS_INTEGRATION: false
Traceback (most recent call last):
  File "manage.py", line 22, in <module>
    execute_from_command_line(sys.argv)
  File "/env3/lib/python3.7/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/env3/lib/python3.7/site-packages/django/core/management/__init__.py", line 357, in execute
    django.setup()
  File "/env3/lib/python3.7/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/env3/lib/python3.7/site-packages/django/apps/registry.py", line 114, in populate
    app_config.import_models()
  File "/env3/lib/python3.7/site-packages/django/apps/config.py", line 211, in import_models
    self.models_module = import_module(models_module_name)
  File "/env3/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/__w/physionet-build/physionet-build/physionet-django/user/models.py", line 22, in <module>
    from project.models import AccessPolicy
  File "/__w/physionet-build/physionet-build/physionet-django/project/models.py", line 1, in <module>
    from project.modelcomponents.access import *
  File "/__w/physionet-build/physionet-build/physionet-django/project/modelcomponents/access.py", line 11, in <module>
    from project.modelcomponents.fields import SafeHTMLField
  File "/__w/physionet-build/physionet-build/physionet-django/project/modelcomponents/fields.py", line 6, in <module>
    from project.utility import LinkFilter
  File "/__w/physionet-build/physionet-build/physionet-django/project/utility.py", line 23, in <module>
    from user.models import User
ImportError: cannot import name 'User' from 'user.models' (/__w/physionet-build/physionet-build/physionet-django/user/models.py)
Error: Process completed with exit code 1.
```

**Comment by fiedosiukr** [comment]
@tompollard I get a ton of emails every time something fails :smile:. There was a circular import. I think I don't need any additional permissions, I am able to view all of the logs.

File: physionet-django/user/models.py [hunk 1]
```diff
@@ -764,14 +765,15 @@ class CredentialApplication(models.Model):
...
-    reference_response_text = models.CharField(max_length=2000,
-        validators=[validators.validate_reference_response])
-    research_summary = models.CharField(max_length=1000,
-        validators=[validators.validate_research_summary])
-    project_of_interest = models.ForeignKey('project.PublishedProject', null=True,
-        on_delete=models.SET_NULL, limit_choices_to={'access_policy': 2},)
+    reference_response = models.PositiveSmallIntegerField(default=0, choices=REFERENCE_RESPONSES)
+    reference_response_text = models.CharField(max_length=2000, validators=[validators.validate_reference_response])
+    research_summary = models.CharField(max_length=1000, validators=[validators.validate_research_summary])
+    project_of_interest = models.ForeignKey(
```
Comment (tompollard) [inline physionet-django/user/models.py:772]: I don't *think* we use the `project_of_interest` field any more, but for consistency maybe `limit_choices_to={'access_policy': AccessPolicy.CREDENTIALED.value}` (plus the import)?

**Review by tompollard — COMMENTED** [review]
Looks great, this is much more readable. The old syntax still appears in places, e.g. as noted inline for user/models.py and: 

https://github.com/MIT-LCP/physionet-build/blob/0c7458ea2929350f33822426d4621f62bf9f1cb7/physionet-django/project/views.py#L1719

https://github.com/MIT-LCP/physionet-build/blob/0c7458ea2929350f33822426d4621f62bf9f1cb7/physionet-django/project/templates/project/published_project_unauthorized.html#L3

I'm good with merging as is, but if you have a chance to switch them beforehand then I'd appreciate it!

**Comment by fiedosiukr** [comment]
@tompollard Fixed.

**Comment by tompollard** [comment]
looks good to me, thanks!

**Pull request merged**